analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion.RoslynCfg.cs

```diff
@@ -79,15 +79,13 @@ public void CheckPaths()
 
                 protected override bool IsValid(BasicBlock block)
                 {
-                    foreach (var operation in block.OperationsAndBranchValue.ToReversedExecutionOrder())
+                    if (block.OperationsAndBranchValue.ToReversedExecutionOrder().FirstOrDefault(x => context.AnalyzedSymbol.Equals(MemberSymbol(x.Instance))) is { } operation)
```

Comment:
Review note: 
- before, there was an iteration over all operations in the block
- inside the loop, if the operation instance is a recursive call, it returns a result

this is much more straight forward now

Reply:
@cristian-ambrosini-sonarsource - was this refactoring necessary to fix the FP? from what I understand, it didn't change anything

and it is very difficult to review IMO (it would have deserved a comment on the PR to say there was no functional change, and in a separate commit)

Reply:
You are right, the refactoring was not related to the issue indeed. In those C#11 sprint, I try to fix code smell/refactor a bit before start working on the issue; I should have put this change in a different commit